build/pluginctl/logs.go

```diff
@@ -0,0 +1,160 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+const (
+	logsPerPage     = 100 // logsPerPage is the number of log entries to fetch per API call
+	timeStampFormat = "2006-01-02 15:04:05.000 Z07:00"
+)
+
+// logs fetches the latest 500 log entries from Mattermost,
+// and prints only the ones related to the plugin to stdout.
+func logs(ctx context.Context, client *model.Client4, pluginID string) error {
+	for i := 5 - 1; i >= 0; i-- {
+		logs, err := fetchLogs(ctx, client, i, pluginID, time.Unix(0, 0))
+		if err != nil {
+			return fmt.Errorf("failed to fetch log entries: %w", err)
+		}
+
+		err = printLogEntries(logs)
+		if err != nil {
+			return fmt.Errorf("failed to print logs entries: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// watchLogs fetches log entries from Mattermost and print them to stdout.
+// It will return without an error when ctx is canceled.
+func watchLogs(ctx context.Context, client *model.Client4, pluginID string) error {
+	now := time.Now()
+	var oldest string
+
+	ticker := time.NewTicker(1 * time.Second)
```

Comment:
Fetching the logs every second feels good from a UX perspective. I think the load is acceptable, but I'm open to increasing the interval.

It could even go down to 500 milliseconds, at least for local dev environments.